### ISSUE TITLE
Jupyter notebook does not save

### ISSUE THREAD

**Issue by ad12** [microsoft/vscode#129190]
Issue Type: <b>Bug</b>

I have a large notebook (~10MB) (not created from but edited in the native interface) that cannot be saved .

Repro:
1. Ctrl + S

VS Code version: Code 1.58.2 (c3f126316369cd610563c75b1b1725e0679adfb3, 2021-07-14T22:09:06.581Z)
OS version: Darwin x64 20.4.0
Restricted Mode: No
Remote OS version: Linux x64 5.8.0-59-generic
Remote OS version: Linux x64 5.8.0-36-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz (12 x 2900)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 3, 3|
|Memory (System)|32.00GB (0.20GB free)|
|Process Argv|--crash-reporter-id 1572910a-3d68-41d5-8b2d-a8224b00f95b|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: torino|
|OS|Linux x64 5.8.0-59-generic|
|CPUs|Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz (32 x 2591)|
|Memory (System)|125.80GB (9.36GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: stelvio|
|OS|Linux x64 5.8.0-36-generic|
|CPUs|Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz (20 x 1200)|
|Memory (System)|125.50GB (13.95GB free)|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
remote-containers|ms-|0.187.1
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
remote-wsl|ms-|0.58.2
vscode-remote-extensionpack|ms-|0.21.0
gc-excelviewer|Gra|3.0.42
python|ms-|2021.7.1053846006
vscode-pylance|ms-|2021.7.4
jupyter|ms-|2021.8.1054968649

</details>

**Comment by rebornix** [comment]
@ad12 can you share a bit more about what happened after you run ctrl+s? Do you get any UI feedback?

**Comment by ad12** [comment]
@rebornix Thanks for your response! There was no UI feedback after pressing ctrl+s - the interface hung for several minutes, after which I restarted VSCode. I wasn't able to run any additional operations on the Jupyter Notebook, so it appeared that the save was blocking any further interaction with the notebook.

I got around this issue by reducing the size of the notebook by removing all cell outputs, but as i run the notebook again, I would like to have the cell outputs persist in the save state to avoid having to recompute them every time i open the notebook.
